### ISSUE TITLE
NPE at InternalPartitionServiceImpl

### ISSUE THREAD

**Issue by pschumacher** [hazelcast/hazelcast#8480]
```
2016-06-29T08:41:52,506+0200 WARN  [hz._hzInstance_1_xxx] com.hazelcast.logging.Slf4jFactory$Slf4jLogger.log(Slf4jFactory$Slf4jLogger.java:90)
.20.30.224:5701 -xxx .5.5
java.lang.NullPointerException: null
	at com.hazelcast.partition.impl.InternalPartitionServiceImpl$RepartitioningTask.run(InternalPartitionServiceImpl.java:1647)
	at com.hazelcast.partition.impl.InternalPartitionServiceImpl$MigrationThread.processTask(InternalPartitionServiceImpl.java:1896)
	at com.hazelcast.partition.impl.InternalPartitionServiceImpl$MigrationThread.doRun(InternalPartitionServiceImpl.java:1872)
	at com.hazelcast.partition.impl.InternalPartitionServiceImpl$MigrationThread.run(InternalPartitionServiceImpl.java:1849)
```

version 3.5.5

**Comment by mdogan** [comment]
@pschumacher: 
- Which Hazelcast version?
- When/how did you see this?

**Comment by pschumacher** [comment]
3.5.5 - forgot to mention, sorry. Unfortunately i just found that in the logs and i have no further information how this could happen.

**Comment by mdogan** [comment]
@pschumacher: thanks for the report.

This where we see NPE in 3.5.5:

https://github.com/hazelcast/hazelcast/blob/v3.5.5/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java#L1647

```
for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
            Address[] replicas = newState[partitionId];             // Line: 1647  <---------- NPE HERE 
            InternalPartitionImpl currentPartition = partitions[partitionId];
            Address currentOwner = currentPartition.getOwnerOrNull();
            Address newOwner = replicas[0];
...
```

Apparently it's fixed already in 3.6 by commit: 1323666f508e7bfdffb0ec8dde2cf6dde0de130b
>Do not continue partition re-assignment if there is no data member

https://github.com/hazelcast/hazelcast/blob/v3.6/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java#L1863

```
Address[][] newState = psg.reArrange(memberGroups, partitions);

if (newState == null) { // <------- HERE NPE CHECK ADDED
    if (logger.isFinestEnabled()) {
        logger.finest("partition rearrangement couldn't be done. size of member groups: " + memberGroups.size());
    }
    return;
}
```

**Comment by pschumacher** [comment]
Thanks for the feedback!
